libs/utils/handleRequestMessage.ts

```diff
@@ -0,0 +1,128 @@
+import { Request, RequestInterface } from "@/libs/models";
+import { callCENNZ } from "@/libs/utils/callCENNZ";
+import { callEthereum } from "@/libs/utils/callEthereum";
+import { fetchRequestDetails } from "@/libs/utils/fetchRequestDetails";
+import { getLogger } from "@/libs/utils/getLogger";
+import { requeueMessage } from "@/libs/utils/requeueMessage";
+import { Api } from "@cennznet/api";
+import { AMQPMessage, AMQPQueue } from "@cloudamqp/amqp-client";
+import { ethers } from "ethers";
+
+const logger = getLogger("RequestProccessor");
+
+export const handleRequestMessage = async (
+	cennzApi: Api,
+	ethersProvider: ethers.providers.BaseProvider,
+	queue: AMQPQueue,
+	message: AMQPMessage,
+	abortSignal: AbortSignal
+): Promise<void> => {
+	let updateRequestRecord: any = null;
+	const body = message.bodyString();
+	if (!body) return;
+	const requestId = Number(body);
+	let messageDelivered = false;
+	updateRequestRecord = createRequestRecordUpdater(requestId) as ReturnType<
+		typeof createRequestRecordUpdater
+	>;
+
+	try {
+		abortSignal.addEventListener(
+			"abort",
+			async () => {
+				if (messageDelivered) return;
+				await updateRequestRecord?.({ status: "Aborted" });
+				await message.reject(false);
+				logger.info("Request #%d: aborted.", requestId);
+			},
+			{ once: true }
+		);
+
+		await updateRequestRecord({
+			status: "Pending",
+			state: "Created",
+		});
+
+		// 1. Fetch request details from CENNZnet
+		if (abortSignal.aborted) return;
+		logger.info("Request #%d: [1/3] fetching details...", requestId);
+		const requestDetails = await fetchRequestDetails(cennzApi, requestId);
+		if (!requestDetails) {
+			await updateRequestRecord({
+				status: "Skipped",
+			});
+			messageDelivered = true;
+			await message.ack();
+			logger.info("Request #%d: skipped.", requestId);
+			return;
+		}
+
+		await updateRequestRecord({
+			requestDetails,
+			state: "InfoFetched",
+		});
+
+		// 2. Call Ethereum with the request details above
+		if (abortSignal.aborted) return;
+		logger.info("Request #%d: [2/3] calling Ethereum...", requestId);
+		const { returnData, blockNumber, blockTimestamp } = await callEthereum(
+			ethersProvider,
+			requestDetails.destination,
+			requestDetails.inputData
+		);
+
+		await updateRequestRecord({
+			state: "EthCalled",
+			ethBlockNumber: blockNumber.toString(),
+		});
+
+		// 3. Submit the `returnData` back to requester
+		if (abortSignal.aborted) return;
+		logger.info("Request #%d: [3/3] calling CENNZnet...", requestId);
+		const result = await callCENNZ(
```

Comment:
can we change the name to something more aligned with the function... like submitStateOrcaleResponseOnCENNZnet

Reply:
yeah good call, I update the name to make it a bit clearer, but probably not that long :)